Add ability to edit and add questions with issue tags.

airyclam opened the pull request:
## Changelist
• Changed precommit script to use package typescript compiler instead of global
• Fixed typescript errors related to input components
• Added a new issueTags query to get all possible issue tags
• Added UI Select component to select from possible issues which is saved to local state as an array and is rendered as clickable boxes which delete from the state when clicked. 

## Out of scope
• Both displaying the issue tags in the candidate guides view and being able to prepopulate the form with existing issue tags would need changes to QuestionResult type generated by the backend, or some sort of logic to parse IssueTagResult as the actual issue labels in the frontend.


<img width="1274" alt="Screenshot 2024-07-03 at 1 02 26 AM" src="https://github.com/populist-vote/web/assets/53859607/f27537e5-2a99-4eb7-bdd8-af2a6c415279">

 
<img width="1300" alt="Screenshot 2024-07-03 at 1 01 27 AM" src="https://github.com/populist-vote/web/assets/53859607/11d8f52f-a4a9-4cb5-959a-cce63ba523ce">

![image (10)](https://github.com/populist-vote/web/assets/53859607/97589a40-b3cd-400a-90d4-95f8cd54d5b1)

airyclam commented:
Related to https://github.com/populist-vote/web/issues/283

wileymc commented on components/QuestionForm/QuestionForm.tsx:
```diff
@@ -62,6 +72,23 @@ function QuestionFormInner({
+      id: string;
+      label: string;
+    }[]
+  >([]);
```
We will want to populate this state using existing question issue tags if they exist as this form is also used to edit existing questions

wileymc replied:
Seeing your note about this—not a blocker

wileymc commented on components/QuestionForm/QuestionForm.tsx:
```diff
@@ -143,6 +189,43 @@ function QuestionFormInner({
+            padding: "1rem 0",
+          }}
+        >
+          {selectedLabels.map(function (label) {
```
just a style consistency nit here but we use arrow functions for callbacks like this throughout the codebase

airyclam replied:
sure I'll make a note to fix this when implementing the other changes for prepopulating and displaying.

wileymc reviewed (changes requested):
Just one change requested: add in a default issue tag state that is populated by querying the existing question for its issue tags

airyclam commented:
## R2 Changelist
• Addressed style comment
• Added ability to view tags from candidate guide page
• Added prepopulated fields

<img width="1293" alt="Screenshot 2024-07-07 at 2 42 48 PM" src="https://github.com/populist-vote/web/assets/53859607/be836c9d-47a6-4dc1-b7f4-ad1be2696473">


https://github.com/populist-vote/web/assets/53859607/f48fe210-dffe-424d-9065-566743d7c862

airyclam commented:
Note: there's a bug here where submitting with only a subset doesn't clear the database entries. Believe this is because the way the data is modeled right now, the table is questionid: issue_tag. Either we have to refactor the query to search and then remove all entries related to questionid when submitting or we have to make the table questionid: [issue_tag1, issuetag2]
